plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {
+  Typography,
+  Grid,
+  Link,
+  Table,
+  TableContainer,
+  TableRow,
+  TableBody,
+  TableHead,
+  TableCell,
+  Paper,
+  Box,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+} from '@material-ui/core';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { InfoCard } from '@backstage/core-components';
+import { NextEscalationPolicyQuery } from './query';
+import QueryQontract from '../../common/QueryAppInterface';
+
+
+const appInterfaceBaseURL = `https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data`;
+
+const getAppInterfaceLink = (path: string) => {
+  return `${appInterfaceBaseURL}${path}`;
+};
+
+export const EscalationPolicyRow = ({ep}: {ep: any}) => {
+  return (  
+    <TableRow>
+      <TableCell width="20%">
+        <Typography>{ep.name}</Typography>
+      </TableCell>
+      <TableCell width="55%">
+        <Typography>{ep.description}</Typography>
+      </TableCell>
+      <TableCell width="25%">
+        <Grid item>
+          <Grid container direction="column">
+            <Grid item>
+              <Typography >
+                <Box sx={{ alignItems: "baseline" }}>
+                  email:
+                  {" "}
+                  <Link target="_blank" href={"mailto:"+ep.channels.email}>
+                    {ep.channels.email}
```

Comment:
This code makes me a bit nervous. You are referring in JSX to nested properties of an object that comes from a data source you don't control. What if one of these properties is missing? Or, what if the `ep.channels.slackUserGroup[0]` array is empty? You are trying to call `ep.channels.slackUserGroup[0].handle`. There's a lot of cases here where this JSX could fail to render and the component could bomb out. I think you need to refactor this to safely pull the properties you want out, and handle cases where stuff isn't there.

Reply:
I refactored the email, slack and jira entries to be instances of a "ContactItem". It makes things easier to read and maintain. Along with that I added a few checks to make sure things exist before using them.